plugins/destination_testing.go

```diff
@@ -3,243 +3,241 @@ package plugins
-		panic(err)
-	}
-	inetColumn := &schema.Inet{}
-	if err := inetColumn.Set("192.0.2.1/24"); err != nil {
-		panic(err)
-	}
-	inetArrayColumn := &schema.InetArray{}
-	if err := inetArrayColumn.Set([]string{"192.0.2.1/24", "192.0.2.1/24"}); err != nil {
-		panic(err)
-	}
-	cidrColumn := &schema.CIDR{}
-	if err := cidrColumn.Set("192.0.2.1"); err != nil {
-		panic(err)
-	}
-	cidrArrayColumn := &schema.CIDRArray{}
-	if err := cidrArrayColumn.Set([]string{"192.0.2.1", "192.0.2.1"}); err != nil {
-		panic(err)
-	}
-	macaddrColumn := &schema.Macaddr{}
-	if err := macaddrColumn.Set("aa:bb:cc:dd:ee:ff"); err != nil {
-		panic(err)
-	}
-	macaddrArrayColumn := &schema.MacaddrArray{}
-	if err := macaddrArrayColumn.Set([]string{"aa:bb:cc:dd:ee:ff", "11:22:33:44:55:66"}); err != nil {
-		panic(err)
-	}
-
-	data := schema.CQTypes{
-		cqID,
-		cqParentID,
-		boolColumn,
-		intColumn,
-		floatColumn,
-		uuidColumn,
-		textColumn,
-		byteaColumn,
-		textArrayColumn,
-		intArrayColumn,
-		timestampColumn,
-		jsonColumn,
-		uuidArrayColumn,
-		inetColumn,
-		inetArrayColumn,
-		cidrColumn,
-		cidrArrayColumn,
-		macaddrColumn,
-		macaddrArrayColumn,
-	}
-
-	return data
+func getTestLogger(t *testing.T) zerolog.Logger {
+	t.Helper()
+	zerolog.TimeFieldFormat = zerolog.TimeFormatUnixMs
+	return zerolog.New(zerolog.NewTestWriter(t)).Output(
+		zerolog.ConsoleWriter{Out: os.Stderr, TimeFormat: time.StampMicro},
+	).Level(zerolog.DebugLevel).With().Timestamp().Logger()
 }
 
-func DestinationPluginTestHelper(ctx context.Context, p *DestinationPlugin, logger zerolog.Logger, spec specs.Destination) error {
+func (s *DestinationPluginTestSuite) destinationPluginTestWriteOverwrite(ctx context.Context, p *DestinationPlugin, logger zerolog.Logger, spec specs.Destination) error {
+	if s.skipTestOverwrite {
+		return nil
+	}
+	spec.WriteMode = specs.WriteModeOverwrite
 	if err := p.Init(ctx, logger, spec); err != nil {
-		return err
+		return fmt.Errorf("failed to init plugin: %w", err)
 	}
+	tableName := "cq_test_write_overwrite"
+	table := testdata.TestTable(tableName)
+	syncTime := time.Now().UTC()
 	tables := []*schema.Table{
-		TestTable(),
+		table,
 	}
 	if err := p.Migrate(ctx, tables); err != nil {
-		return err
+		return fmt.Errorf("failed to migrate tables: %w", err)
+	}
+
+	sourceName := uuid.NewString()
+	resource := schema.DestinationResource{
+		TableName: table.Name,
+		Data:      testdata.TestData(),
+	}
+	resource2 := schema.DestinationResource{
+		TableName: table.Name,
+		Data:      testdata.TestData(),
+	}
+	_ = resource2.Data[5].Set("00000000-0000-0000-0000-000000000007")
+	resources := []schema.DestinationResource{
+		resource,
+		resource2,
 	}
-	syncTime := time.Now()
-	sourceName := "test_helper_" + syncTime.String()
 
-	resources := make(chan *schema.DestinationResource, 1)
-	expectedResource := &schema.DestinationResource{
-		TableName: tables[0].Name,
-		Data:      TestData(),
+	if !s.skipTestOverWriteDeleteStale {
```

Comment:
These lines don't impact the outcome of the test (you can remove them and `TestDestinationPlugin` passes

Reply:
Not sure I understand. some plugins dont support `DeleteStale`

Reply:
I mean you can remove the entire block (including `DeleteStale`) and the test still passes

Reply:
But I want to run deletestale for plugins that support it so why remove it? or we are talking about a different block.

Reply:
That is what's confusing for me, why do we need it if the test passes without it?

Reply:
I removed it I was referring to a deletestale lateron where I check if it was deleted after that via read